### PR TITLE
[demand] Escape special characters in item name

### PR DESCRIPTION
**Pull request by fproldan**
https://github.com/agritheory/beam/issues/200

### CODE REVIEW

File: beam/beam/demand/demand.py [hunk 1]
```diff
@@ -632,106 +676,137 @@ def get_descendant_warehouses(company: str, warehouse: str) -> list[str]:
...
-		ORDER BY delivery_date, idx, creation, parent ASC
-	"""
+	if a_filters:
+		allocation_query = allocation_query.where(*a_filters)
 
 	record_offset = records_per_page * (page - 1)
-	query = (
-		f"{demand_query} UNION ALL {allocation_query} LIMIT {records_per_page} OFFSET {record_offset}"
-	)
+	final_query = demand_query.union(allocation_query).limit(records_per_page).offset(record_offset)
```
Comment (fproldan) [inline beam/beam/demand/demand.py]: TO FIX: individual queries works, the union doesn't

Comment (agritheory) [reply beam/beam/demand/demand.py]: @fproldan Heather figured out a strategy for this in Check Run where we needed some queries to be configurable

https://github.com/agritheory/check_run/blob/version-14/check_run/check_run/doctype/check_run/check_run.py#L702-L719

**Comment by fproldan** [comment]
The tests are passing locally

Captura de pantalla 2024-10-11 a la(s) 2 27 34 p  m

**Review by agritheory — APPROVED** [review]
@fproldan This looks great! Demand map builds with the customer data where we discovered the problem.

**Pull request merged**